### ISSUE TITLE
Border\Overscan Area

### ISSUE THREAD

**Issue by mjj03301977** [MiSTer-devel/NES_MiSTer#235]
Pretty sure default behavior for Famicom/NES is showing the border/overscan area which shows a combination of a solid color (depending on game) and possibly garbage data. Shouldn't this be default rather than hiding all of this with a black border? Would it be possible to just add an option for black border area instead of having it be default which is not true to the original system?

**Comment by Kitrinx** [comment]
Crop overscan is off by default, which will show the entire 240 lines vertically.

That being said, it is painfully obvious from the extremely distracting distortion which is regularly present in this area that the system was designed in a way expecting this area to be masked by the overscan of the television. When snes came out, the television of the day were showing more of the screen so they blacked out that area intentionally in case the tv didn't crop enough.

That said, I'd argue that the more authentic way to view NES is with cropping the overscan on. Nonetheless the crop is off by default.

**Comment by mjj03301977** [comment]
Crop is indeed off by default, but the border area is black. This is not authentic. Is there anyway to set the core to seen the true border color along with graphical garbage? Maybe there is and I'm just not seeing it?

**Comment by Kitrinx** [comment]
Not at present, I assume you are talking about the horizontal border. There's actually no garbage there per se, in the NTSC version the background color is drawn there (or the address if you set it to vram address during non-draw time). PAL consoles mask it black. It's extremely ugly and distracting, and no games use it which is why almost no emulators show it. This also wouldn't have been very visible on a real television. The greyscale line on the side in these screenshots is authentic to real hardware, so that's part of it. Currently the ppu doesn't generate this area in a useable way.

![line2](https://user-images.githubusercontent.com/15653832/108885253-93051600-75d5-11eb-836c-0c0d8be97dbe.PNG)
![greystripe](https://user-images.githubusercontent.com/15653832/108885261-95677000-75d5-11eb-909f-709f73036bc6.PNG)

**Comment by mjj03301977** [comment]
Yeah! That's what I'm asking for...at least a setting to toggle this mode off/on. :)

**Comment by Kitrinx** [comment]
In a refactor of the ppu I'm doing I have this for debugging. If I get this refactor in shape to merge I will add this option with it. The current ppu can't really do it correctly.

**Comment by mjj03301977** [comment]
Awesome, thank you!!

Hoping for your "MATH" palette too! ;)

**Comment by retrorepair** [comment]
> In a refactor of the ppu I'm doing I have this for debugging. If I get this refactor in shape to merge I will add this option with it. The current ppu can't really do it correctly.

Did you get round to this?

Ive noticed the AR compared to a PAL NES seems off too, like the start of a scanline is drawn too early/padding isn't present but it's wider overall on a CRT as the image on the right matches up? I'll try to get aome captures to show what I mean (it may be a separate issue to be honest)

**Comment by mjj03301977** [comment]
This issue has been fixed! Thank you so much Kitrinx! :)
